### DISCUSSION TITLE
Networking stack status

### DISCUSSION THREAD

**Discussion by iangrunert** [oxidecomputer/hubris#294] [General]
What's the status on the networking stack for hubris?

I see there's a proposal open for the networking stack: https://github.com/oxidecomputer/hubris/pull/98
and an basic ethernet implementation for the stm32h7 open: https://github.com/oxidecomputer/hubris/pull/158

Any rough finger-in-the-wind estimate on when those might land? Is there opportunity to help contribute in that area?

**Comment by cbiffle** [comment]
Hi! My initial prototype netstack has landed as of this week. It might be more limited than you'd hope:

- Currently it's focused on the stm32h743-nucleo board. We'll be generalizing it to other stm32h7 boards. Currently we don't have a need to generalize it to non-stm32h7 MACs.
- We're currently using `smoltcp` for the actual IP processing.
- We're focused on IPv6-only.
- We're not implementing TCP (despite `smoltcp`'s name, it's quite a capable UDP stack as well).

However, it might still be interesting to look at, or could serve as a jumping-off point for other work. Possibly its most interesting characteristic, to me, is that we know the total set of required sockets at compile time, so we don't have to allocate them. The source is here:

- net task: https://github.com/oxidecomputer/hubris/tree/master/task/net
- stm32h7 MAC driver: https://github.com/oxidecomputer/hubris/tree/master/drv/stm32h7-eth
- example server: https://github.com/oxidecomputer/hubris/tree/master/task/udpecho
